extensions/wikia/CommunityPage/CommunityPageSpecialController.class.php

```diff
@@ -0,0 +1,243 @@
+			'allMembers' => $this->msg( 'communitypage-view-all-members' )->plain(),
+			'recentlyJoinedHeaderText' => $this->msg( 'communitypage-recently-joined' )->plain(),
+			'members' => $recentlyJoined,
+		] );
+	}
+
+	/**
+	 * Set context for allMembers template. Needs to be passed through the index method in order to work.
+	 * @return array
+	 */
+	public function getAllMembersData() {
+		$allMembers = $this->usersModel->getAllMembers();
+
+		$this->response->setData( [
+			'allMembersHeaderText' => $this->msg( 'communitypage-all-members' )->plain(),
+			'admin' => $this->msg( 'communitypage-admin' )->plain(),
+			'joinedText' => $this->msg( 'communitypage-joined' )->plain(),
+			'members' => $allMembers,
+		] );
+	}
+
+	/**
+	 * Set context for recentActivityModule template. Needs to be passed through the index method in order to work.
+	 * @return array
+	 */
+	protected function getRecentActivityData() {
+		$data = $this->sendRequest('LatestActivityController', 'executeIndex')->getData();
+		$recentActivity = [];
+
+		foreach ($data['changeList'] as $activity) {
+			$changeType = $activity['changetype'];
+
+			switch ($changeType) {
+				case 'new':
+					$changeTypeString = $this->msg( 'communitypage-created' )->plain();
+					break;
+				case 'delete':
+					$changeTypeString = $this->msg( 'communitypage-deleted' )->plain();
+					break;
+				case 'edit':
+					// fall through
+				default:
+					$changeTypeString = $this->msg( 'communitypage-edited' )->plain();
+					break;
+			}
+
+			$changeMessage = $this->msg( 'communitypage-activity',
+				$activity['user_href'], $changeTypeString, $activity['page_href'] )->plain();
+
+			$recentActivity[] = [
+				'timeAgo' => $activity['time_ago'],
+				'userAvatar' => AvatarService::renderAvatar(
+					$activity['user_name'],
+					AvatarService::AVATAR_SIZE_SMALL_PLUS ),
+				'userName' => $activity['user_name'],
+				'userHref' => $activity['user_href'],
+				'changeTypeString' => $changeTypeString,
+				'editedPageTitle' => $activity['page_title'],
+				'changeMessage' => $changeMessage,
+			];
+		}
+
+		$title = SpecialPage::getTitleFor( 'WikiActivity' );
+
+		return [
+			'activityHeading' => $data['moduleHeader'],
+			'moreActivityText' => $this->msg( 'communitypage-recent-activity' )->plain(),
+			'moreActivityLink' => $title->getCanonicalURL(),
+			'activity' => $recentActivity,
+		];
+	}
+
+	public function getModalHeaderData() {
+		$adminData =  $this->sendRequest( 'CommunityPageSpecialController', 'getTopAdminsData' )->getData();
+		$memberCount = $this->usersModel->getMemberCount();
+
+		$this->response->setData([
+			'allText' => $this->msg( 'communitypage-modal-tab-all' )->plain(),
+			'allCount' => $memberCount,
+			'adminsText' => $this->msg( 'communitypage-modal-tab-admins' )->plain(),
+			'adminsCount' => $adminData['adminCount'],
+			'leaderboardText' => $this->msg( 'communitypage-modal-tab-leaderboard' )->plain(),
+		]);
+	}
+
+	protected function addAssets() {
```

Comment:
As a general OOP guideline, it's best to default to private unless otherwise needed so the class is better encapsulated and you can change the internal implementation without affecting anything that uses or extends your class. Protected should be used for things you know you want to be accessible/overridable by subclasses, and it can always be switched to protected if there is a need to extend the class.